internal/provider/provider.go

```diff
@@ -1936,11 +1936,20 @@ func assumeRoleSchema() *schema.Schema {
 		MaxItems: 1,
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
+				"duration": {
+					Type:          schema.TypeString,
+					Optional:      true,
+					Description:   "The duration of the role session, e.g. 1h. Valid time units are ns, us (or µs), ms, s, h, or m.",
+					ValidateFunc:  verify.ValidTimeDuration,
```

Comment:
Q for reviewer: This validation does not allow for empty strings...or should it?

Reply:
Looking at the validation functions provided by the Plugin SDK, and especially [`validation.IsRFC3339Time()`](https://github.com/hashicorp/terraform-plugin-sdk/blob/36ee3aa4bc4b9a4fa75ccf85b481293fa51c1ecc/helper/validation/time.go#L42), we're probably fine not checking for it.